lib/serviceBuilder.js

```diff
@@ -45,11 +43,21 @@ function convertBody(body) {
 }
 
 function makeCallAsStream(endUrl, method, body, options) {
-  return sgetPromisified({
-    url: endUrl,
-    method,
-    body,
-    headers: options.headers,
+  return new Promise(function promiseFunction(resolve, reject) {
+    sget({
+      url: endUrl,
+      method,
+      body,
+      headers: options.headers,
+    }, function onResult(error, res) {
+      if (error) {
+        return reject(error)
+      }
+      if (options.allowedStatusCodes && !options.allowedStatusCodes.includes(res.statusCode)) {
+        return reject(new Error('Status code of the response is not included in the allowed status codes.'))
```

Comment:
'Status code of the response is not included in the allowed status codes.' => 'Invalid status code: "". Allowed: ""